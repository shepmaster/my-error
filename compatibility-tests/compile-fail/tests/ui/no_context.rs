extern crate snafu;

use snafu::Snafu;

type BoxError = Box<dyn std::error::Error>;

#[derive(Debug, Snafu)]
enum Error {
    #[snafu(context(false))]
    MissingSource {},

    #[snafu(context(false))]
    HasUserFields { source: BoxError, a: i32, b: i32 },
}

fn main() {}
