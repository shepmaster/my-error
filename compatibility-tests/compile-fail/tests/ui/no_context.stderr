error: Context selectors without context must not have context fields
  --> $DIR/no_context.rs:13:39
   |
13 |     HasUserFields { source: BoxError, a: i32, b: i32 },
   |                                       ^^^^^^

error: Context selectors without context must not have context fields
  --> $DIR/no_context.rs:13:47
   |
13 |     HasUserFields { source: BoxError, a: i32, b: i32 },
   |                                               ^^^^^^

error: Context selectors without context must have a source field
  --> $DIR/no_context.rs:10:5
   |
10 |     MissingSource {},
   |     ^^^^^^^^^^^^^
